File: templates/template.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

class template {

  public static void part1(ArrayList<String> lines) {
    long solution = 0;

    System.out.println("Part 1: " + solution);
  }

  public static void part2(ArrayList<String> lines) {
    long solution = 0;

    System.out.println("Part 2: " + solution);
  }

  public static void main(String[] args) {
    ArrayList<String> lines = new ArrayList<>();

    if (args.length < 1) {
      System.err.println("Invalid Usage. Usage: java template [filename]");
      System.exit(1);
    }

    try (Scanner scn = new Scanner(new File(args[0]))) {
      while (scn.hasNextLine()) {
        lines.add(scn.nextLine());
      }
      part1(lines);
      part2(lines);
    } catch (FileNotFoundException e) {
      e.printStackTrace();
      System.err.println("File not found: " + args[0]);
    }

    System.exit(0);
  }

}

File: templates/template.js
function part1(lines) {
  let ans = 0;

  console.log(`Part 1: ${ans}`);
}

function part2(lines) {
  let ans = 0;

  console.log(`Part 2: ${ans}`);
}

const fs = require('fs');

if (process.argv.length < 3) {
  console.error('Invalid Usage.\nUsage: node template.js [input file]');
  process.exit(1);
}

fs.readFile(process.argv[2], 'ascii', (err, data) => {
  if (err) throw err;
  part1([...data]);
  part2([...data]);
});
